java/code/src/com/redhat/rhn/manager/setup/ProductSyncManager.java

```diff
@@ -495,10 +492,11 @@ private SetupWizardProductDto convertProduct(final MgrSyncProductDto productIn)
      * Find a product for any given ident by looking through base and their addons.
      *
      * @param ident ident of a product
+     * @param productsWithAddons the products list with addons
      * @return the {@link SetupWizardProductDto}
      */
-    private SetupWizardProductDto findProductByIdent(String ident) {
-        for (SetupWizardProductDto p : getBaseProducts()) {
+    private SetupWizardProductDto findProductByIdent(String ident, List<SetupWizardProductDto> productsWithAddons) {
+        for (SetupWizardProductDto p : productsWithAddons) {
```

Comment:
we still iterate over the whole list to find the ident. But I think we do not safe so much time if we convert it into a map using ident as key and the object as value to get a direct access.
So ok for me.